### PR TITLE
feat: display incompatibility message for LNS swap, LIVE-14891, LIVE-14838, LIVE-14923

### PR DESCRIPTION
**Pull request by andreicovaciu**
### ✅ Checklist

- [x] `npx changeset` was attached.
- [ ] **Covered by automatic tests.** 
- [ ] **Impact of the changes:** 
  - ...

### 📝 Description

Adds LNS incompatibility banner for TON, ADA and SPL Tokens.
image (6)

### ❓ Context

- **JIRA or GitHub link**: LIVE-14838 

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

### CODE REVIEW

**Review by Justkant — CHANGES_REQUESTED** [review]
I think we might want to handle the `renderHardwareUpdate` before the `DeviceAction` component to avoid sending APDUs to the device while showing the message

File: apps/ledger-live-desktop/src/renderer/components/DeviceAction/index.tsx [hunk 1]
```diff
@@ -341,7 +341,12 @@ export const DeviceActionDefaultRendering = <R, H extends States, P>({
   }
 
   if (device?.modelId === "nanoS" && (request as SwapRequest)?.provider === "thorswap") {
-    return renderHardwareUpdate();
+    return (
+      <HardwareUpdate
+        i18nKeyTitle="swap.wrongDevice.title"
+        i18nKeyDescription="swap.wrongDevice.description"
+      />
+    );
```
Comment (Justkant) [inline apps/ledger-live-desktop/src/renderer/components/DeviceAction/index.tsx:349]: We also want to move this part as it's working in the same way and for swap too

File: apps/ledger-live-desktop/src/renderer/components/LiveAppDrawer.tsx [hunk 2]
```diff
@@ -54,6 +57,14 @@ export type StartExchangeData = {
   onResult: (startExchangeResult: StartExchangeSuccessResult) => void;
 };
 
+type SwapRequest = {
+  transaction: Transaction;
+  exchange: ExchangeSwap;
+  provider: string;
+  rate: number;
+  amountExpectedTo: number;
+};
```
Comment (Justkant) [inline apps/ledger-live-desktop/src/renderer/components/LiveAppDrawer.tsx:66]: It doesn't seem like we use this type

Comment (Justkant) [reply apps/ledger-live-desktop/src/renderer/components/LiveAppDrawer.tsx:66]:
```suggestion
```

File: apps/ledger-live-desktop/src/renderer/components/LiveAppDrawer.tsx [hunk 3]
```diff
@@ -54,6 +57,7 @@ export type StartExchangeData = {
   onResult: (startExchangeResult: StartExchangeSuccessResult) => void;
 };
 
+
```
Comment (Justkant) [inline apps/ledger-live-desktop/src/renderer/components/LiveAppDrawer.tsx:60]:
```suggestion
```

**Review by Justkant — DISMISSED** [review]
I've tested a lot of cases with multiple devices and it now works perfectly, thx for the updates

**Review by Justkant — APPROVED** [review]

**Review by jiyuzhuang — APPROVED** [review]

**Pull request merged**